Vagrantfile

```diff
@@ -0,0 +1,74 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+
+Vagrant.configure(2) do |config|
+  config.vm.box = "ubuntu/trusty64"
+  host_port = ENV["GIRDER_HOST_PORT"] || 8080
+
+  sync_folders = ENV["DEVELOPMENT_SYNC_FOLDERS"] || false
+
+  config.vm.network "forwarded_port", guest: 8080, host: host_port
+
+  config.vm.define "minerva" do |node| end
+
+  config.vm.provider "virtualbox" do |vb|
+    host = RbConfig::CONFIG['host_os']
+
+    # Give VM 1/4 system memory & access to 1/2 of the cpu cores on the host
+    if host =~ /darwin/
+      cpus = `sysctl -n hw.ncpu`.to_i
+      cpus = cpus / 2
+      # sysctl returns Bytes and we need to convert to MB
+      mem = `sysctl -n hw.memsize`.to_i / 1024 / 1024 / 4
+    elsif host =~ /linux/
+      cpus = `nproc`.to_i
+      cpus = cpus / 2
+      # meminfo shows KB and we need to convert to MB
+      mem = `grep 'MemTotal' /proc/meminfo | sed -e 's/MemTotal://' -e 's/ kB//'`.to_i / 1024 / 4
+    else # Guess!
+      cpus = 2
+      mem = 4096
+    end
+
+    vb.customize ["modifyvm", :id, "--memory", mem]
+    vb.customize ["modifyvm", :id, "--cpus", cpus]
+    vb.customize ["modifyvm", :id, "--nictype1", "virtio"]
+    vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+    vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
+
+  end
+
+
+  config.vm.synced_folder ".", "/vagrant", disabled: true
+
+  if sync_folders
+    if File.directory?("../girder")
+      config.vm.synced_folder "../girder", "/opt/girder", owner: 1003, group: 1003
```

Comment:
Can synced_folders take strings of the user/group rather than the magic numbers? Or a comment indicating what they are, I'm assuming vagrant/vagrant?